docs/proposals/28-autonomous-shoot-clusters.md

```diff
@@ -0,0 +1,497 @@
+- [`gardenadm init`](#gardenadm-init) to bootstrap a first control plane node.
+- [`gardenadm token`](#gardenadm-token) to manage bootstrap and discovery tokens for `gardenadm join`.
+- [`gardenadm join`](#gardenadm-join) to bootstrap further control plane nodes, or to bootstrap worker nodes and join them to the cluster.
+- [`gardenadm connect`](#gardenadm-connect) to deploy a `gardenlet` for further cluster management.
+- [`gardenadm version`](#gardenadm-version) to print the `gardenadm` version.
+
+The general flow for bootstrapping an autonomous shoot cluster will be as follows:
+
+1. The user installs `gardenadm` on the first control plane node and runs `gardenadm discover`.
+   This will download relevant configuration from the Gardener system the autonomous shoot cluster shall be later connected to (e.g., `Controller{Registration,Deployment}`s, `CloudProfile`s, etc.).
+   If there is no existing Gardener system yet, the needed configuration must be provided manually.
+2. The downloaded configuration and a `Shoot` manifest are provided to the `gardenadm init` command.
+   This will bootstrap the first control plane node.
+3. Optionally, additional control plane nodes can be joined.
+   In order to do so, the user runs `gardenadm token generate` which generates a random bootstrap token, and then `gardenadm token create <token>` which creates it on the server.
+   Now connected to the new node that shall be joined, the user runs the `gardenadm join` command which bootstraps another control plane node.
+4. Optionally, worker nodes can be joined.
+   The flow is the same - a bootstrap token must be generated and created, and then `gardenadm join` must be executed to bootstrap the worker node.
+5. Once above steps are completed, the user runs `gardenadm connect` to deploy a `gardenlet` into the cluster which connects to an existing Gardener system for further cluster management.
+
+A more detailed description of the commands is following.
+
+#### `gardenadm discover`
+
+This command takes a kubeconfig to an existing Gardener system, and a `Shoot` manifest.
+It downloads the required `CloudProfile`, `ControllerRegistration` and `ControllerDeployment` resources and stores them locally for usage of other `gardenadm` commands.
+This simplifies the crafting of these documents, and it also ensures that the configuration and versions used for the bootstrapping of the autonomous shoot cluster fit to the configuration later found in the connected Gardener system.
+
+#### `gardenadm init`
+
+This command bootstraps the first control plane node of the autonomous shoot cluster.
+The configuration is specified via separate files.
+It needs a `Shoot` manifest and some additional configuration files, which usually reside in the garden cluster, e.g., a `CloudProfile` (best downloaded with `gardenadm discover`).
```

Comment:
It will be used in the same way like it's done for regular shoots - validation of `Shoot` configuration as well as usage of central configuration (e.g., for keystone URLs for OpenStack). Why do you have the impression that this is different/should be highlight specially?

Reply:
I was mostly wondering about the high touch scenario and whether the `CloudProfile` is needed at all there, as the user is responsible to create most (if not all) of the infrastructure resources (if I understand [this line correctly](https://github.com/gardener/gardener/blob/99eefb454c4ebd6701eab79312bee3ddd8ae1f56/docs/proposals/28-autonomous-shoot-clusters.md?plain=1#L332)).

Reply:
Yes, maybe you're right. Perhaps it's not needed in this scenario, but today, the `CloudProfile` is a hard requirement for all shoots. We have to dive into this once implementing this GEP, and when appropriate, lift this requirement.